spec/unit/recipes/disks_spec.rb

```diff
@@ -7,24 +7,47 @@
 
 describe 'mapr::disks' do
   context 'When all attributes are default, on centos 7.4.1708' do
+    let(:disk0) { '/dev/disk/by-id/wwn-0x12345678' }
+    let(:disk2) { '/dev/disk/by-id/wwn-0x02346578' }
+    let(:disk3) { '/dev/disk/by-id/wwn-0x00346578' }
     let(:chef_run) do
       runner = ChefSpec::SoloRunner.new(
         platform:  'centos',
         version:   '7.4.1708',
         step_into: %w[mapr_disksetup],
       ) do |node|
-        node.override['mapr']['mfs']['disks'] = %w[disk1 disk2 disk3]
+        ['z', 'y', 't'].each do |t|
+          node.override['block_device']["sd#{t}"]['state'] = t == 'y' ? 'dead' : 'running'
+          node.override['block_device']["sd#{t}"]['removable'] = '0'
+        end
       end
       runner.converge(described_recipe)
     end
 
+    before do
+      disk1 = '/dev/disk/by-id/wwn-0x12345eff'
+      disk4 = '/dev/disk/by-id/wwn-0x5000c500921464d9'
+      disk5 = '/dev/disk/by-id/wwn-0x50005c00921464d9'
+
+      allow(::Dir).to receive(:glob).and_call_original
+      allow(::Dir).to receive(:glob).with(['/dev/disk/by-id/wwn-*']).and_return [disk4, disk0, disk3, disk1, "#{disk1}-part1", disk2]
+
+      allow(::File).to receive(:readlink).and_call_original
+      allow(::File).to receive(:readlink).with(disk0).and_return '../../sdz'
+      allow(::File).to receive(:readlink).with(disk2).and_return '../../sdz'
+      allow(::File).to receive(:readlink).with(disk1).and_return '../../sdt'
+      allow(::File).to receive(:readlink).with(disk3).and_return '../../sdt'
```

Comment:
Normal that disk3 and disk1 have different serials and map to the same disk ? I think you may need to explicit what you're trying to prove here because it's not clear

Reply:
You are right. I have thus made all the disks unique